docs/docs/rfcs/00000000_add_resource_conditionally.md

```diff
@@ -0,0 +1,33 @@
+- Feature Name: Add resources conditionally
+- Status: in-progress
+- Start Date: 2022-04-08
+- Authors: Vikash & Sushmith
+
+# Summary
+
+Resource in Guardian represents the actual resource in the provider e.g. for BigQuery provider, a resource represents a
+dataset or a table. Guardian collects resources from the provider automatically as soon as it registered. While in
+parallel, Guardian also has a job for continuously syncing resources.
+
+In current design it fetch all the database/table/resources in configure provider, there is no way to filter out on
+resources.
+
+# Technical Design
+
+Add a scope of condition in provider configuration to filter out on resources.
+
+#### Using expression
+
+Expression can be used to check whether Guardian should add the resources or not.
+
+```yaml
+resources:
+  add_if: $resource.urn endsWith 'keyset'
```

Comment:
can we rename this field to `filter`?

Reply:
@rahmatrhd sure, what you suggest for this filter name

Reply:
@singhvikash11 `filter` itself.

Reply:
Wouldn't `add_if` be more appropriate here? `filter` can mean either `filtering out` or `include these only`. `add_if` wouldn't have that confusion.

Reply:
`filter` is fine I guess and can be utilized for generic purpose, or we can use `where` to make it more concise/

Reply:
@bsushmith the `filter` field will have expression value which expected to return boolean, from there we can infer whether the resource will be included or filtered out. Same as javascript's `Array.filter()` api